src/math/matrix.c

```diff
@@ -3,93 +3,115 @@
 // Copyright(c) 2022 Intel Corporation. All rights reserved.
 //
 // Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+//	   Shriram Shastry <malladi.sastry@linux.intel.com>
+//
 
 #include <sof/math/matrix.h>
 #include <errno.h>
 #include <stdint.h>
 
-int mat_multiply(struct mat_matrix_16b *a, struct mat_matrix_16b *b, struct mat_matrix_16b *c)
+/* Performs matrix multiplication of two fixed-point 16-bit integer matrices,
+ * storing the result in a third matrix. It accounts for fractional bits for
+ * fixed-point arithmetic, adjusting the result accordingly.
+ *
+ * Arguments:
+ *   a: pointer to the first input matrix
+ *   b: pointer to the second input matrix
+ *   c: pointer to the output matrix to store result
+ *
+ * Return:
+ *   0 on successful multiplication.
+ *   -EINVAL if input dimensions do not allow for multiplication.
+ *   -ERANGE if the shift operation might cause integer overflow.
+ */
```

Comment:
Adding documentation is good, but using the doxygen syntax would be better.

Reply:
Done